public/app/features/trails/utils.ts

```diff
@@ -115,3 +118,42 @@ export function getFilters(scene: SceneObject) {
   }
   return null;
 }
+
+// frontend hardening limit
+const MAX_ADHOC_VARIABLE_OPTIONS = 10000;
+
+export function limitAdhocProviders(
```

Comment:
Also would be nice if we add some comment lines about what this function is about. This is for our future selves :)

Reply:
Adding comments!